File: flow-client/src/main/java/com/vaadin/client/flow/reactive/ReactiveValueChangeListener.java
/*
 * Copyright (C) 2000-2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See <https://vaadin.com/commercial-license-and-service-terms> for the full
 * license.
 */
package com.vaadin.client.flow.reactive;

/**
 * Listens to changes to a reactive value.
 *
 * @see ReactiveValue#addReactiveValueChangeListener(ReactiveValueChangeListener)
 *
 * @author Vaadin Ltd
 * @since 1.0
 */
@FunctionalInterface
public interface ReactiveValueChangeListener {
    /**
     * Invoked when a reactive value has changed.
     *
     * @param event
     *            the change event
     */
    void onValueChange(ReactiveValueChangeEvent event);
}

File: flow-client/src/main/frontend/internal/client/flow/reactive/ReactiveValueChangeListener.ts
/*
 * Copyright 2000-2026 Vaadin Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

// TypeScript port of com.vaadin.client.flow.reactive.ReactiveValueChangeListener.

import type { ReactiveValueChangeEvent } from './ReactiveValueChangeEvent';

/**
 * Listens to changes to a reactive value.
 *
 * @see {@link ReactiveValue.addReactiveValueChangeListener}
 */
export type ReactiveValueChangeListener = (event: ReactiveValueChangeEvent) => void;
